### ISSUE TITLE
Serenity/JS 3.0 - Report graph errors due to missing bootstrap minified files for datatables v1.10.20

### ISSUE THREAD

**Issue by maxe2y** [serenity-js/serenity-js#1234]
When reporting using Serenity-js version 3.0.0-rc.19, the folder generated does not include the bootstrap datatables files minified. Instead, it includes the following folder tree
![image](https://user-images.githubusercontent.com/55079984/174766543-7058923f-46e5-42b2-93cc-e58a99d25a50.png)

when the report search for the bootstrap version 1.10.20 it fails 
<img width="1672" alt="image" src="https://user-images.githubusercontent.com/55079984/174789678-bc8a2ee9-1b24-4034-8efb-e89415c9c4d3.png">

**Comment by viper3400** [comment]
Hi @maxe2y, I use this version of Serenity/JS on my own and my report works fine. So maybe you can provide a bit more context of your project setup? 

What [Serenity/JS modules](https://serenity-js.org/modules/) do you use? 
I assume, this report has worked before for you. With which version did it worked the last time for you?

It's hard to tell what's the problem here, without knowing more. 

Would you be able to create a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) in a repo on GitHub? I tried to reproduce the issue using the [cucumber template](https://github.com/serenity-js/serenity-js-cucumber-webdriverio-template). 

* clone the repo
* switch to the `main` branch
* delete the package-lock.json (because of some old references in there)
* run `npm install`
* run `npm test`

In my case the report was created including all graphs.

**Comment by maxe2y** [comment]
Yes sure
It worked for latest version 2

I'm using `yarn test ` in this project
https://github.com/maxe2y/Serenity_V3

you can check on your local it uses the webdriver template , but with Serenity V3 applied
the report is uploaded and you can check each file generated.

In the index.html file lines 87-88
[image](https://user-images.githubusercontent.com/55079984/174997701-91071b43-17a4-4bd0-94c5-5a8acaf9a300.png)

it requires datatables dependencies that do not exist in the report folder as I mentioned before
also there are more dependecies error in the browser console 

![image](https://user-images.githubusercontent.com/55079984/174998197-b0d382df-9ac6-4328-9950-233558020c6e.png)

**Comment by viper3400** [comment]
I was able to reproduce this with your repo, partly, though the project is taken from the 'master' branch, which is still on 2.x and does not include the changes, as it would be on the 'main' branch. So tests won't run at all.

But I think I got a least a trace. Do you have something like

```
[test:execute] [0-0] error: unknown option '--strict'
```

in your console log when running `yarn test`?

Can you check your `wdio.conf.ts` and set `strict: false` in the `cucumberOpts`?

**Comment by viper3400** [comment]
> Do you have a fully functional branch of webdriver with serenity 3 

At least I thought so ;) 

Just use the `main` branch of the template instead of the master: https://github.com/serenity-js/serenity-js-cucumber-webdriverio-template/tree/main

Then delete the `package-lock.json` and I would assume, that a `yarn install` and `yarn test` should work, as well.

**Comment by viper3400** [comment]
If you’d like to rule out cucumber, there would be also an example using `mocha`: https://github.com/serenity-js/serenity-js-mocha-webdriverio-template/tree/main

**Comment by maxe2y** [comment]
https://github.com/serenity-js/serenity-js-cucumber-webdriverio-template/compare/main...maxe2y:main
failing the same way same lines

**Comment by maxe2y** [comment]
<img width="1627" alt="image" src="https://user-images.githubusercontent.com/55079984/175103661-1aa35ba0-98f1-4cab-98ea-6e3d88c2439f.png"> 🤔 
can you confirm this text is printed out in the scenario on your side too?

**Comment by viper3400** [comment]
I will try out again 👍

**Comment by viper3400** [comment]
Hi @maxe2y,
so, just to be sure, [this](https://github.com/maxe2y/serenity-js-cucumber-webdriverio-template/tree/main/target/site/serenity) is the report you've generated and the last screenshot is taken from this report, right?

Because then it's getting even more weird:

If I just clone your repo, switch to the `main` branch containing the report, open it locally in Firefox (or Chrome, doesn't matter) I see a report that looks quite good to me. Don't can confirm the issues from your screenshot, though I assume we're looking at the same report. 🤔 

![grafik](https://user-images.githubusercontent.com/11006334/175346354-14f62e69-d2d6-4471-90fc-f08c9e856a5c.png)

As provided by the report, you are on Mac OS X, right. I'm on Windows 11, and not used to Mac OS at all. Could it be, that some css stuff or something is blocked or old stuff is cached in your browser cache.

**Comment by maxe2y** [comment]
Great!
Now just focus in the main issue we are talking about
"Report graph errors due to missing bootstrap minified files for datatables v1.10.20"
those errors appears in the dashboard and the dependency is missing
Did you check the index.html file generated?

**Comment by viper3400** [comment]
So, one step closer - I have nearly the same errors in the dashboard. 
No difference between the files, you've provided and when I would generate the report on my own.

Though, my dashboard looks ok. 

![grafik](https://user-images.githubusercontent.com/11006334/175990688-3be71898-4424-440f-85d1-71d42d30e1cf.png)

What is missing on my side is the very first error which can be seen in your screenshot from the issue description:

```
Refused to apply style from '<URL>' because its MIME type ('text/html') is not a supported stylesheet MIME type, and strict MIME checking is enabled..
```

Do you access the report from a file, directly, or from a web server?

@jan-molak, can you please check? This is most likely an issue with the serenity-bdd reporter, right?  Have a look here:  https://github.com/serenity-bdd/serenity-core/issues/1971 

I  already tried latest [Serenity-BDD cli 3.2.5](https://mvnrepository.com/artifact/net.serenity-bdd/serenity-cli) using the [artifact parameter](https://github.com/serenity-js/serenity-js/issues/1133#issuecomment-1043494086) - but with that I got even more errors in the console log ... 🤔 



(Just to make the relation: Update of the cli is blocked by https://github.com/serenity-js/serenity-js/issues/1147 anyway).

**Comment by viper3400** [comment]
@maxe2y , as a workaround, you could always try to apply the [artifact parameter](https://github.com/serenity-js/serenity-js/issues/1133#issuecomment-1043494086) to force usage of an older version, aka the version of the bdd cli which worked for you before with Serenity/JS 2.x

**Comment by maxe2y** [comment]
I'm not migrating to Serenity V3 until is stable 👍

**Comment by jan-molak** [comment]
@maxe2y - out of curiosity, could you please try with Serenity BDD CLI version 3.2.5?

```
"test:update-serenity": "serenity-bdd update --ignoreSSL --artifact=net.serenity-bdd:serenity-cli:jar:3.2.5",
"test:report": "serenity-bdd run --artifact=net.serenity-bdd:serenity-cli:jar:3.2.5"
```

**Comment by maxe2y** [comment]
I will be a try

**Comment by maxe2y** [comment]
Yup 
looks better
![image](https://user-images.githubusercontent.com/55079984/176422524-445dd967-9712-45de-b426-8d937678ec95.png)

**Comment by jan-molak** [comment]
Great, thanks for confirming @maxe2y. Please let me know if you spot any errors.

I'm doing some exploratory testing with it now, but if neither of us find any issues I'll update Serenity/JS 3.0.0-rc to use Serenity BDD 3.2.5

**Comment by viper3400** [comment]
@jan-molak, please wait with the update. Remember, there is still this pending issue https://github.com/serenity-js/serenity-js/issues/1147

I can retry with latest 3.2.5 version though and give feedback.

**Comment by viper3400** [comment]
@jan-molak , issue with mocha specs is still not solved in 3.2.5, see #1147

**Comment by jan-molak** [comment]
Thanks for confirming @aukevanleeuwen, glad 3.2.5 is working for you.

@viper3400 - let's finish #1236 and see what can be done about #1147.

**Comment by jan-molak** [comment]
@aukevanleeuwen - it looks like the issue with the broken report might have been caused by a bug fixed in c5ca1bf and released in [3.1.4](https://serenity-js.org/changelog/3.1.4).

Serenity BDD does not escape HTML tags present in the `json` reports, so Serenity/JS had to escape them before handing the data over to Serenity BDD.

Please let me know if you come across this (or any other issue) again. Thanks!

**Comment by maxe2y** [comment]
release **3.15.0** fixes the issues with css, now that it comes with **serenity-cli-4.0.43**.
